frontend/src/main/webui/src/components/Job.jsx

```diff
@@ -28,28 +27,54 @@ const JobRow = ({tag, value, style}) => (
 let stateUpdateTimer;
 
 class Job extends React.Component {
-
-
     constructor(props) {
         super(props);
+        this.state = {
+            showModal: false,
+            buttonPressed: -1,
```

Comment:
good for now, but in js it's better to use string identifiers for state, it's more explicit than just numbers. Here the var name could be 'outputType' and its states 'inputVolume', 'outputVolume' or 'console'